lib/osml/model_runner/roles/mr_task_role.ts

```diff
@@ -74,72 +85,206 @@ export class MRTaskRole extends Construct {
       ),
-      managedPolicies: [
-        ManagedPolicy.fromAwsManagedPolicyName("AmazonSQSFullAccess"),
-        ManagedPolicy.fromAwsManagedPolicyName("AmazonS3FullAccess"),
-        ManagedPolicy.fromAwsManagedPolicyName("AmazonDynamoDBFullAccess"),
-        ManagedPolicy.fromAwsManagedPolicyName("AmazonSageMakerFullAccess"),
-        ManagedPolicy.fromAwsManagedPolicyName("CloudWatchFullAccess"),
-        ManagedPolicy.fromAwsManagedPolicyName("SecretsManagerReadWrite"),
-        ManagedPolicy.fromAwsManagedPolicyName(
-          "AmazonElasticContainerRegistryPublicFullAccess"
-        )
-      ],
       description:
         "Allows the Oversight Model Runner to access necessary AWS services (S3, SQS, DynamoDB, ...)"
     });
 
+    const mrPolicy = new ManagedPolicy(this, "MRTaskPolicy", {
+      managedPolicyName: "MRTaskPolicy"
+    });
+
     // Add permissions to assume roles
-    mrTaskRole.addToPolicy(
-      new PolicyStatement({
-        effect: Effect.ALLOW,
-        actions: ["sts:AssumeRole"],
-        resources: ["*"]
-      })
-    );
+    const stsPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["sts:AssumeRole"],
+      resources: ["*"]
+    });
 
     // Add permissions for AWS Key Management Service (KMS)
-    mrTaskRole.addToPolicy(
-      new PolicyStatement({
-        effect: Effect.ALLOW,
-        actions: ["kms:Decrypt", "kms:GenerateDataKey", "kms:Encrypt"],
-        resources: [
-          `arn:${this.partition}:kms:${props.account.region}:${props.account.id}:key/*`
-        ]
-      })
-    );
-
-    // Add permissions for AWS Events
-    mrTaskRole.addToPolicy(
-      new PolicyStatement({
-        actions: ["events:PutRule", "events:PutTargets", "events:DescribeRule"],
-        resources: [
-          `arn:${this.partition}:events:${props.account.region}:${props.account.id}:*`
-        ],
-        effect: Effect.ALLOW
-      })
-    );
+    const kmsPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["kms:Decrypt", "kms:GenerateDataKey", "kms:Encrypt"],
+      resources: [
+        `arn:${this.partition}:kms:${props.account.region}:${props.account.id}:key/*`
+      ]
+    });
 
     // Add permissions for Amazon Kinesis
-    mrTaskRole.addToPolicy(
-      new PolicyStatement({
-        effect: Effect.ALLOW,
-        actions: ["kinesis:PutRecord", "kinesis:PutRecords"],
-        resources: [
-          `arn:${this.partition}:kinesis:${props.account.region}:${props.account.id}:stream/*`
-        ]
-      })
-    );
+    const kinesisPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        "kinesis:PutRecord",
+        "kinesis:PutRecords",
+        "kinesis:DescribeStream"
+      ],
+      resources: [
+        `arn:${this.partition}:kinesis:${props.account.region}:${props.account.id}:stream/${this.mrSyncConfig.KINESIS_RESULTS_STREAM}-${props.account.id}`
```

Comment:
How does this work if we don't have a Kinesis sync defined or we don't enable that output sync?

Reply:
Updated it to use `asterisks`. Even if Kinesis ARN is not provided, it will use the alternative output method which is S3.